### DISCUSSION TITLE
Make gradient in dark mode `background-image` work with simple background `background-color` in light mode?

### DISCUSSION THREAD

**Discussion by deadcoder0904** [tailwindlabs/tailwindcss#5123] [Help] [answered]
I have `bg-blue-100 text-blue-900 dark:bg-info` where `bg-info` is `linear-gradient(to right, hsla(240, 100%, 50%, 0.2), transparent 50%)`

I get the light mode working fine but `dark:bg-info` doesn't work as it uses `background-image` CSS property rather than `background-color` CSS property for light mode.

How do I make the dark mode work?

I've made a Stackblitz demo showing the problem -> https://stackblitz.com/edit/github-6frqvs-tqnsnl?file=pages%2Findex.tsx

Open in New Window as Tailwind Dark Mode doesn't show side-by-side on Stackblitz.

Notice, how it doesn't show blue gradient on Dark Mode despite the code being:

```html
<div className="p-4 m-4 h-20 w-96 bg-red-400 text-red-900 dark:bg-info">Hello hi</div>
```

**Comment by deadcoder0904** [comment, marked as answer]
I've managed to make Dark Mode work on Tailwind Play so here's the demo -> https://play.tailwindcss.com/fecClyOaIZ

Check only the 1st box. 2nd box is a demonstration of how it must really look in Dark Mode.

In Dark Mode, the 1st box should look like dark blue with a black gradient (see 2nd box)

# Light Mode (1st box looks correct in light mode)

![light-mode](https://user-images.githubusercontent.com/16436270/127772617-cccbb7eb-49c1-48ff-b520-c518d557cba0.png)

# Dark Mode (1st box looks incorrect in dark mode. Should actually look like 2nd box)

![dark-mode](https://user-images.githubusercontent.com/16436270/127772626-5026b20a-788a-4b11-971d-ad8c93b56dda.png)

@adamwathan Is there any solution to this?

**Reply by huibizhang** [reply]
Okay is me, again XD

css can allow background-color and background-image at same time.
so in Tailwind , that is the reason why box1 is incorrect.

just give box1 a dark:bg-transparent, it's set background-color to transparent

https://play.tailwindcss.com/D4kUJTP6VF

**Reply by huibizhang** [reply]
based on this trick, can play something interesting, like blend background-image and background-color together.

see [background blend mode](https://tailwindcss.com/docs/background-blend-mode) of tailwind docs

**Reply by deadcoder0904** [reply]
@huibizhang Damn, man you've solutions to every problem ahaha. Works perfectly well :)

**Reply by deadcoder0904** [reply]
@huibizhang I don't understand blend mode. I just tried doing `dark:bg-blend-color-burn` but doesn't give anything :(

**Reply by huibizhang** [reply]
@deadcoder0904 umm ... I don't know how to explain it ! XD
the effect will depending on the actual colors.

for example. mode: darken
if upper layer is red, lower layer is white,
red is darker than white, so white will be replaced by red you see.
instead, blend-mode lighten will let black getting brighter and red you see.

and,  if you using background-color and background-image at same time,
color is upper layer, and image is lower layer

you can see this example:
https://play.tailwindcss.com/syOVmbsnmC
https://play.tailwindcss.com/gWNjvmNOS3

**Reply by deadcoder0904** [reply]
@huibizhang yeah, i've seen it's applications on images which look quite decent but for everything else it's not that great so i'll leave it for images only :)

**Reply by simonswiss** [reply]
Thanks for all the help, @huibizhang ✨
